### ISSUE TITLE
Implement a weather system

### ISSUE THREAD

**Issue by CurryKitten** [CurryKitten/CurryKitten-Sim#74]
Considering the idea of a weather system.  Just because the nice permanently sunny daytime doesn't accurately represent that many areas of the world.  This could allow people to fly in more overcast conditions, rain (if they are crazy) or night.  Although my thoughts are this is a cosmetic thing as opposed to being blown around in a gale.

**Comment by CurryKitten** [comment]
Just about finished this - I just need to sort out the cameras that work on the LOS versions of the quad/plane as we manipulate the shadow bias in order to make it easier to see if you model if close to the ground... and that means I need to know where the lighting comes from (which changes depending on the weather... or the use of it)

I've just finished the rather mammoth task of implementing a UI to control all of this, and there's a reasonable amount to manipulate.  The obvious one is if you want the weather system or just continue as is.  As running all those shaders is quite graphically intensive, this will murder older machines, hence the option to turn them off, but there's also a lower graphics version which will use 2d, instead of volumetric clouds and not cast shadows from them.  Then we have all the fun of deciding when you fly, whether time will flow and if so how long it'll take for a day/night cycle.  After all of that, I needed to make sure that any changed settings would save and restore correctly on startup.

![Screenshot 2021-01-23 at 00 39 15](https://user-images.githubusercontent.com/12443664/105562895-f4567280-5d13-11eb-8668-b7db84b0e53a.jpg)
